crates/networking/discv5/src/config.rs

```diff
@@ -1,17 +1,95 @@
+        || u32::from_be_bytes(addr.octets()) == 0xc000000a
+    {
+        return true;
+    }
+    !addr.is_private()
+            && !addr.is_loopback()
+            && !addr.is_link_local()
+            && !addr.is_broadcast()
+            && !addr.is_documentation()
+            // shared
+            && !(addr.octets()[0] == 100 && (addr.octets()[1] & 0b1100_0000 == 0b0100_0000)) &&!(addr.octets()[0] & 240 == 240 && !addr.is_broadcast())
+            // addresses reserved for future protocols (`192.0.0.0/24`)
+            // reserved
+            && !(addr.octets()[0] == 192 && addr.octets()[1] == 0 && addr.octets()[2] == 0)
+            // Make sure the address is not in 0.0.0.0/8
+            && addr.octets()[0] != 0
+}
+
+impl Default for NetworkConfig {
+    fn default() -> Self {
+        let mut subnets = Subnets::new();
+        // Enable attestation subnets 0 and 1 as a reasonable default
+        let _ = subnets.enable_subnet(Subnet::Attestation(0));
+        let _ = subnets.enable_subnet(Subnet::Attestation(1));
+
+        let filter_rate_limiter = Some(
+            discv5::RateLimiterBuilder::new()
+                .total_n_every(10, Duration::from_secs(1)) // Allow bursts, average 10 per second
+                .ip_n_every(9, Duration::from_secs(1)) // Allow bursts, average 9 per second
+                .node_n_every(8, Duration::from_secs(1)) // Allow bursts, average 8 per second
+                .build()
+                .expect("The total rate limit has been specified"),
+        );
+
+        let discv5_listen_config =
+            discv5::ListenConfig::from_ip(Ipv4Addr::UNSPECIFIED.into(), 9000);
+
+        // discv5 configuration
+        let discv5_config = discv5::ConfigBuilder::new(discv5_listen_config)
+            .enable_packet_filter()
+            .session_cache_capacity(5000)
+            .request_timeout(Duration::from_secs(2))
+            .query_peer_timeout(Duration::from_secs(2))
+            .query_timeout(Duration::from_secs(30))
+            .request_retries(1)
+            .enr_peer_update_min(10)
+            .query_parallelism(8)
+            .disable_report_discovered_peers()
+            .ip_limit() // limits /24 IP's in buckets.
+            .incoming_bucket_limit(8) // half the bucket size
+            .filter_rate_limiter(filter_rate_limiter)
+            .filter_max_bans_per_ip(Some(5))
+            .filter_max_nodes_per_ip(Some(10))
+            .table_filter(|enr| enr.ip4().is_some_and(|ip| is_global_ipv4(&ip))) // Filter non-global IPs
+            .ban_duration(Some(Duration::from_secs(3600)))
+            .ping_interval(Duration::from_secs(300))
+            .build();
+
+        Self {
+            discv5_config,
+            bootnodes: Vec::new(),
+            socket_address: IpAddr::V4(std::net::Ipv4Addr::new(0, 0, 0, 0)),
+            socket_port: 9000,
+            disable_discovery: false,
+            total_peers: 10,
+            subnets,
+        }
+    }
```

Comment:
can this be removed

Reply:
i did this to give Default implementation for Config , it is being used for tests and can be utilised for code examples to test module.
 let me know if you are not fine with this, this code is mostly taken from lighthouse :) I should give them credit for this.

Reply:
I think you should remove it then

Reply:
ok sure, should i remove tests as well as i need default values for tests , or i can give test values in the module but mostly same values will be given , Default trait implementation was added for common test values.

Reply:
removed `is_global_ipv4` function and trimmed down the `Default` code to initiate `Config`,  i can move it to test module if it is not required.